### PR TITLE
NEW Add rule to catch malformed _t() calls

### PR DESCRIPTION
**Pull request by GuySartorelli**
This should catch cases where keys in calls to `_t()` (either globally or via `i18n::_t()` directly) are malformed. It'll also catch some cases where the text collector wouldn't be able to correctly evaluate the key.

## Issue
- https://github.com/silverstripe/silverstripe-standards/issues/5

### CODE REVIEW

File: tests/PHPStan/TranslationFunctionRuleTest/InClass.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        _t('abc');
+        _t(InClass::class);
+        _t(InClass::class . 'somekey');
+        _t(InClass::MY_ENTITY);
+        _t(__CLASS__ . InClass::MY_ENTITY);
+        $variable = 'somethingwrong';
+        _t($variable);
+        i18n::_t('nodot');
+        $callableString = '_t';
+        $callableString('nodot');
```
Comment (emteknetnz) [inline tests/PHPStan/TranslationFunctionRuleTest/InClass.php:22]: If you're going to this degree of detection, you may as well also include call_user_func() and call_user_func_array()

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: tests/PHPStan/TranslationFunctionRuleTest/InClass.php (hunk 1 above)
Comment (ChloeSartorelli) [reply tests/PHPStan/TranslationFunctionRuleTest/InClass.php:22]: PHPStan doesn't treat those the same way, so that would require adding a bunch of code for an extreme edge case. I'd rather not do that, especially given text collector won't identify those anyway.

**Review by emteknetnz — APPROVED** [review]
